source/MenuUtils.cpp

```diff
@@ -37,6 +38,129 @@ std::string string_format(const std::string &format, Args... args) {
     return std::string(buf.get(), buf.get() + size - 1); // We don't want the '\0' inside
 }
 
+uint32_t remapWiiMoteButtons(uint32_t buttons) {
+    uint32_t convButtons = 0;
+
+    if (buttons & WPAD_BUTTON_LEFT)
+        convButtons |= VPAD_BUTTON_LEFT;
+
+    if (buttons & WPAD_BUTTON_RIGHT)
+        convButtons |= VPAD_BUTTON_RIGHT;
+
+    if (buttons & WPAD_BUTTON_DOWN)
+        convButtons |= VPAD_BUTTON_DOWN;
+
+    if (buttons & WPAD_BUTTON_UP)
+        convButtons |= VPAD_BUTTON_UP;
+
+    if (buttons & WPAD_BUTTON_PLUS)
+        convButtons |= VPAD_BUTTON_PLUS;
+
+    if (buttons & WPAD_BUTTON_2)
+        convButtons |= VPAD_BUTTON_Y;
+
+    if (buttons & WPAD_BUTTON_1)
+        convButtons |= VPAD_BUTTON_X;
+
+    if (buttons & WPAD_BUTTON_B)
+        convButtons |= VPAD_BUTTON_B;
+
+    if (buttons & WPAD_BUTTON_A)
+        convButtons |= VPAD_BUTTON_A;
+
+    if (buttons & WPAD_BUTTON_MINUS)
+        convButtons |= VPAD_BUTTON_MINUS;
+
+    if (buttons & WPAD_BUTTON_HOME)
+        convButtons |= VPAD_BUTTON_HOME;
+
+    return convButtons;
+}
+
+uint32_t remapClassicButtons(uint32_t buttons) {
+    uint32_t convButtons = 0;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_LEFT)
+        convButtons |= VPAD_BUTTON_LEFT;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_RIGHT)
+        convButtons |= VPAD_BUTTON_RIGHT;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_DOWN)
+        convButtons |= VPAD_BUTTON_DOWN;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_UP)
+        convButtons |= VPAD_BUTTON_UP;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_PLUS)
+        convButtons |= VPAD_BUTTON_PLUS;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_X)
+        convButtons |= VPAD_BUTTON_X;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_Y)
+        convButtons |= VPAD_BUTTON_Y;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_B)
+        convButtons |= VPAD_BUTTON_B;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_A)
+        convButtons |= VPAD_BUTTON_A;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_MINUS)
+        convButtons |= VPAD_BUTTON_MINUS;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_HOME)
+        convButtons |= VPAD_BUTTON_HOME;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_ZR)
+        convButtons |= VPAD_BUTTON_ZR;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_ZL)
+        convButtons |= VPAD_BUTTON_ZL;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_R)
+        convButtons |= VPAD_BUTTON_R;
+
+    if (buttons & WPAD_CLASSIC_BUTTON_L)
+        convButtons |= VPAD_BUTTON_L;
+
+    return convButtons;
+}
+
+InputData getInput() {
+    VPADStatus vpadStatus;
+    VPADReadError vpadError;
+    KPADStatus kpadStatus;
+    KPADError kpadError;
+    InputData inputData;
+
+    VPADRead(VPAD_CHAN_0, &vpadStatus, 1, &vpadError);
```

Comment:
The return value of VPADRead is ignored and `vpadError` is not initalized. `inputData` also needs to be initalized!

Maybe do something like this:

```
VPADStatus vpadStatus {};
KPADStatus kpadStatus {};
InputData inputData {};
KPADError kpadError = KPAD_ERROR_UNINITIALIZED;
VPADReadError vpadError = VPAD_READ_UNINITIALIZED;
if (VPADRead(VPAD_CHAN_0, &vpad, 1, &vpadError) == 1 && vpadError == VPAD_READ_SUCCESS) {
    inputData.trigger = vpadStatus.trigger;
    inputData.hold    = vpadStatus.hold;
    inputData.release = vpadStatus.release;
}
```